### PR TITLE
openvpn: reworked status file parser

### PR DESCRIPTION
**Pull request by rpv-tomsk**
* Added detection of status fields index by parsing 'HEADER' lines.
   This allows to support status files from latest OpenVPN-2.4 and possible from future versions;
 * Fixed a bug with OpenVPN status file version 3 where delimiter is 'tab' (not space or comma);
 * Status file version detection was moved from configuration callback to read callback;
 * Now 'plugin_register_complex_read()' is used for registering read callbacks.

Closes: #2128

### CODE REVIEW

**Comment by rpv-tomsk** [comment]
I have checked this on my setups. Status files of 'SINGLE' , 'MULTI' version 1, version 2, version 3 formats from OpenVPN-2.2; 'MULTI' version 2 status file from OpenVPN-2.4 was checked.

@octo, @rubenk - I offer this already polished PR to your attention. Let's review and merge it?
I will be happy to answer your questions.

**Comment by rpv-tomsk** [comment]
CC @ZdenekBrichacek

File: src/openvpn.c [hunk 1]
```diff
@@ -276,319 +302,198 @@ static int multi1_read(const char *name, FILE *fh) {
...
+  /* OpenVPN-2.4 has 11 fields of data + 2 fields for "HEADER" and "CLIENT_LIST"
+   * So, set array size to 20 elements, to support future extensions.
+   */
+  char *fields[20];
   const int max_fields = STATIC_ARRAY_SIZE(fields);
-  int fields_num, read = 0;
+  int fields_num;
   long long sum_users = 0;
+  
+  int found_header = 0;
```
Comment (octo) [inline src/openvpn.c]: Nit: this should be a `_Bool`

**Comment by rpv-tomsk** [comment]
Hi, Florian! Glad to get your feedback! 

The patch is rebased to current master, 'nits' are fixed, code passed through clang-format.

**Pull request merged**